### PR TITLE
refactor code

### PR DESCRIPTION
**Pull request by Ntare22**
refactor calculator and App to functional components to use hooks

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
+
+  const clickHandler = (e) => {
+    setState((prev) => calculate(prev, e.target.value));
+  };
+
+  const { total, next } = state;
+  const stateOfCalc = { total, next };
+  return (
+    <>
+      <Calculator handleClick={clickHandler} calcState={stateOfCalc} />
```
Comment (D4nielJ) [inline src/App.js]: [OPTIONAL] I think you don't need to deconstruct the object here, to construct a new object to pass on props. You could just pass the state in `calcState={state}` and save a couple of lines. :sweat_smile:

File: src/components/Calculator.js [hunk 2]
```diff
@@ -1,35 +1,23 @@
...
-    const { total, next } = this.state;
-    if (next) {
-      value = next;
-    } else if (total) {
-      value = total;
-    }
+  return (
+    <>
+      <input className="calc-screen" placeholder="0" value={value} readOnly />
+      <NumsOperations handleClick={handleClick} />
```
Comment (D4nielJ) [inline src/components/Calculator.js:18]: - [ ] Let's change NumsOperations to work as a stateless functional component as the requirement of this project is to refactor all components.

**Review by D4nielJ — CHANGES_REQUESTED** [review]
## Hello @Ntare22, 

You're doing great! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights :fire: :rocket: 
- Your application works correctly after the refactor of main components!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

## Happy coding and stay awesome! 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can always find me on slack!

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AshakaE — APPROVED** [review]
### Hi @Ntare22   🙋‍♀️, 

# STATUS APPROVED ✔

Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Ntare22** [comment]
> ### Hi @Ntare22 🙋‍♀️,
> # STATUS APPROVED ✔
> Your project is complete! There is nothing else to say other than... it's time to merge it 🔗 🚢 Congratulations! 🎉
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
> 
> Cheers 🥂 and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Hey..

Thanks for the approval. Can you change the status on the platform, so that I'm able to submit the final project.

**Pull request merged**